tests/models/test_sasrec.py

```diff
@@ -582,3 +593,337 @@ def test_get_dataloader_recommend(
+    ) -> None:
+        item_net = ItemNetConstructor.from_dataset(
+            dataset_item_features, n_factors=n_factors, dropout_rate=0.5, item_net_block_types=item_net_block_types
+        )
+        assert item_net.get_all_embeddings().shape == (item_net.n_items, n_factors)
+
+    @pytest.mark.parametrize(
+        "item_net_block_types,sparse_features,expected_n_item_net_blocks",
+        (
+            ((IdEmbeddingsItemNet,), True, 1),
+            ((IdEmbeddingsItemNet, CatFeaturesItemNet), True, 2),
+            ((IdEmbeddingsItemNet,), False, 1),
+            ((IdEmbeddingsItemNet, CatFeaturesItemNet), False, 1),
+        ),
+    )
+    def test_create_from_dataset(
+        self,
+        dataset_item_features: Dataset,
+        item_net_block_types: tp.Sequence[tp.Type[ItemNetBase]],
+        sparse_features: bool,
+        expected_n_item_net_blocks: int,
+    ) -> None:
+        if not sparse_features:
+            item_features = pd.DataFrame(
+                [
+                    [11, "f3", 0],
+                    [12, "f3", 1],
+                    [13, "f3", 2],
+                    [14, "f3", 3],
+                    [15, "f3", 4],
+                    [17, "f3", 5],
+                    [16, "f3", 6],
+                ],
+                columns=["id", "feature", "value"],
+            )
+            ds = Dataset.construct(
+                INTERACTIONS,
+                item_features_df=item_features,
+            )
+        else:
+            ds = dataset_item_features
+
+        item_net: ItemNetConstructor = ItemNetConstructor.from_dataset(
+            ds, n_factors=10, dropout_rate=0.5, item_net_block_types=item_net_block_types
+        )
+
+        actual_n_items = item_net.n_items
+        actual_item_net_blocks = len(item_net.item_net_blocks)
+
+        assert actual_n_items == dataset_item_features.item_id_map.size
+        assert actual_item_net_blocks == expected_n_item_net_blocks
+
+    @pytest.mark.parametrize(
+        "item_net_block_types,n_items,n_factors",
+        (
+            ((IdEmbeddingsItemNet,), 2, 16),
+            ((IdEmbeddingsItemNet, CatFeaturesItemNet), 4, 8),
+        ),
+    )
+    def test_embedding_shape_after_model_pass(
+        self,
+        dataset_item_features: Dataset,
+        item_net_block_types: tp.Sequence[tp.Type[ItemNetBase]],
+        n_items: int,
+        n_factors: int,
+    ) -> None:
+        items = torch.from_numpy(
+            np.random.choice(dataset_item_features.item_id_map.internal_ids, size=n_items, replace=False)
+        )
+        item_net: ItemNetConstructor = ItemNetConstructor.from_dataset(
+            dataset_item_features, n_factors=n_factors, dropout_rate=0.5, item_net_block_types=item_net_block_types
+        )
+
+        expected_embeddings = item_net(items)
+
+        assert expected_embeddings.shape == (n_items, n_factors)
+
+    def test_raise_when_no_item_net_blocks(self) -> None:
```

Comment:
Let's also add a test where we pass only `CatFeaturesItemNet` block and dataset doesn't have item features or has only dense features or has sparse features without categories.

Reply:
Fixed